### ISSUE TITLE
Blockchain Explorer

### ISSUE THREAD

**Issue by qraz** [forknote/forknote-pool#45]
Blockchain explorer is not displaying the blocks/transactions, it just shows the empty page with no values.

I've setup the pool in a subdirectory like http://xyz.com/pool
And here is my front-end settings

> var api = "http://xyz.com/pool:8117";
> var api_blockexplorer = "http://xyz.com/pool:1118";
> var poolHost = "xyz.com/pool";

And I've added this to the coin config in the pool

> rpc-bind-ip=0.0.0.0
> enable-blockchain-indexes=1
> enable-cors=*

Please advise as to what I need to change to display the blockchain correctly.

**Comment by pmitchev** [comment]
are you able to make RPC to the daemon not from localhost?

**Comment by qraz** [comment]
hmm I'm not sure how to check that. Could you please explain it?

**Comment by qraz** [comment]
My bad, I used a wrong url while testing the RPC method. With the right URL now I was able to RPC from local as well as remote machine (tested  for the method **getblockheaderbyheight** and it returns values as expected).

Any idea what else I could check?

**Comment by servc4** [comment]
Hi
I have same issue...
when I run command from console where the pool is runing
` curl demonhost:18081/json_rpc -d '{"method":"getblockheaderby                                                                                        height","params":{"height":100}}'`

I get answer
`{"jsonrpc":"2.0","result":{"block_header":{"depth":279,"difficulty":79,"hash":"32dd9b678f9c038fcf233c688fea3c25e109ae1b1226bb9c56dcc4c6d0f37dda","height":1443,"major_version":3,"minor_version":0,"nonce":1694,"orphan_status":false,"prev_hash":"537759fb577780e375608ae0c80e4b0d16813caca7db761b32ae50fa0f8f042f","reward":69982455237895,"timestamp":1513276082},"status":"OK"}}`

then in config I set
`var api_blockexplorer = "demonhost:18081";`
and there is no results..
Any idea why ??

**Comment by flarecurrency** [comment]
I got this error

(Thread 1) Error polling getblocktemplate {"code":"ECONNREFUSED","errno":"ECONNREFUSED","syscall":"connect"}

**Comment by flarecurrency** [comment]
I got the samething right now, but when I try this trough the web page its not even loading or something.

curl 127.0.0.1:38985/json_rpc -d '{"method":"getblockheaderbyheight","params":{"height":100}}'

result: 
{"jsonrpc":"2.0","result":{"block_header":{"depth":4711,"difficulty":4228,"hash":"0540de153be718e083038b5be10ffc173250a42a17a947339122fe01282078be","height":99,"major_version":3,"minor_version":0,"nonce":209761323,"orphan_status":false,"prev_hash":"71001a262f179dfdf70ab18c424617ce9c9e7a47ebf8c46ec070254a7dc76e2f","reward":1525302758107,"timestamp":1516014979},"status":"OK"}}

**Comment by servc4** [comment]
Hi 
Forknote block explorer required more commands included in RPC Deamon server. Take a look on my command and answer from deamon is much different then yours. You need to add commands from forknote to your deamon after that hopefully will work.

Regards

**Comment by flarecurrency** [comment]
Hello,

Is the blockchain working by you?

**Comment by servc4** [comment]
Hi 
not all info you can find on it but those are small thins what we need to resolve but is working.
 
regards

**Comment by flarecurrency** [comment]
Hello,

Could you share your code?

**Comment by servc4** [comment]
Hi
It's not mine ;)
I use this one as a block explorer
https://github.com/Karbovanets/Karbowanec-Blockchain-Explorer

Good lock

**Comment by flarecurrency** [comment]
Hello,

It's not the blockexplorer what is wrong, but the code from the pool, because it send the post and I get a response and the pool code don't show the information.

![image](https://user-images.githubusercontent.com/35806256/35644935-a798e6be-06ca-11e8-8bcd-6b06792e7782.png)

**Comment by flarecurrency** [comment]
@servc4 thanks

**Comment by servc4** [comment]
Yes but like you see you ask about block 3333 and respond is with block 3332
That's why i decide to use https://github.com/Karbovanets/Karbowanec-Blockchain-Explorer
it's work.
Regards

**Comment by servc4** [comment]
Hi
I almost lunch properly working explorer but I have issue with CROS header
` Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at http://127.0.0.1:18081/getinfo. (Reason: CORS header ‘Access-Control-Allow-Origin’ missing).`

How I can resolve that problem
.htaccess
```
Header set Access-Control-Allow-Origin "*"
Header add Access-Control-Allow-Headers "origin, x-requested-with, content-type"
Header add Access-Control-Allow-Methods "PUT, GET, POST, DELETE, OPTIONS"
```
Is not working
I need to have ad-don in my browser to view page properly.

Regards

**Comment by flarecurrency** [comment]
@servc4 I fixed the blockchain explorer from the pool.

![screenshot from 2018-02-01 16-11-07](https://user-images.githubusercontent.com/35806256/35685677-a7410246-076a-11e8-89f6-305ad14cace7.png)

You can fix the problem by changing this in your config.js


```
var blockchainExplorer = "http://YOUR POOL ADDRES/?hash={id}#blockchain_block";

var transactionExplorer = "http://YOUR POOL ADDRES/?hash={id}#blockchain_transaction";
```

**Comment by servc4** [comment]
You mean after line 165
 response.addHeader("Content-Type", "application/json");

I need to add this
response.addHeader("Access-Control-Allow-Origin", "*");

because I did that and recompile the source and is same...

Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at http://127.0.0.1:18081/getinfo. (Reason: CORS header ‘Access-Control-Allow-Origin’ missing).

**Comment by hongbaoblockchain** [comment]
@flarecurency where do you upload files to for front end website? Like if im using lampp? Backend code is not an issue for me just not sure of website side.

**Comment by hongbaoblockchain** [comment]
Pool runs but cant get anything to work with api what am i missing. I have pool saved to /var/www/html 127.0.0.1/pool shows website with no data. I been playing around with api and have it set to 127.0.0.1/pool:1118 but nothing. The pools connected to blockchain. I see it making requests to my seed nodes. But website part wont post data. Any help guys?

**Comment by blueavenue** [comment]
It is solved the issue yet?

**Comment by servc4** [comment]
Hi
In file /src/HTTP/HttpResponse.cpp
change:

`HttpResponse::HttpResponse() {
  status = STATUS_200;
  headers["Server"] = "CryptoNote-based HTTP server";
}`

on this:

`HttpResponse::HttpResponse() {
  status = STATUS_200;
  headers["Server"] = "CryptoNote-based HTTP server";
  headers["Access-Control-Allow-Origin"] = "*";
}`

end build source.
Regards

**Comment by blueavenue** [comment]
Thx i will try it and give feedback.

**Comment by 3ch0elite** [comment]
Anyfix with using just forknote-linux? Only issue i have testing is blockdetails not showing i just get page 400 error not found seems to point to 0.0.0.0/cointicker/blockaddress

**Comment by blueavenue** [comment]
Feedback: It doesnt works. I build code again with the code you all put in here. 
I have no Blockchain Explorer. 

What Daemon you use? Forknote with config-file? Forknote without config-file? Or another Coin-Repo? 

Additional Question:

What is the var api_explorer in the website config.js? And what var i have to put in? 

Please help!

**Comment by servc4** [comment]
You need to run your demon with additional
--enable-blockexplorer
in start command

Regards

**Comment by blueavenue** [comment]
Thx servc4 but disnt work. Have put this in the conf-file (like in the forknote-pool install instruction):

rpc-bind-ip=0.0.0.0
enable-blockexplorer=1
enable-cors=*

What you mean with start command? Upstart Service?

**Comment by blueavenue** [comment]
THX to all! All works fine! The Pool is up! 

Only the network site doesnt show me the pool and other informations. But i think, i can solve the issue the next days.

**Comment by inzider** [comment]
> 
> 
> THX to all! All works fine! The Pool is up! (62.75.139.132/pool) - Later with Domainname. (I edit Coin Source [http_server.cpp and rpc_server.cpp] and Build again)
> 
> Only the network site doesnt show me the pool and other informations. But i think, i can solve the issue the next days.

Hi! Do you remember what you did to show the blocks in blockexplorer?
